snippets/visualbasic/VS_Snippets_CLR_Classic/classic SHA256 Example/VB/source.vb

```diff
@@ -2,68 +2,59 @@
 Imports System
 Imports System.IO
 Imports System.Security.Cryptography
-Imports System.Windows.Forms
 
-Public Class HashDirectory
+Public Module HashDirectory
 
-    Public Shared Sub Main(ByVal args() As String)
-        Dim directory As String
+    Public Sub Main(ByVal args() As String)
         If args.Length < 1 Then
-            Dim fdb As New FolderBrowserDialog
-            Dim dr As DialogResult = fdb.ShowDialog()
-            If (dr = DialogResult.OK) Then
-                directory = fdb.SelectedPath
-            Else
-                Console.WriteLine("No directory selected")
-                Return
-            End If
-        Else
-            directory = args(0)
+            Console.WriteLine("No directory selected")
+            Return
         End If
-        Try
+
+        Dim targetDirectory As String = args(0)
+        If Directory.Exists(targetDirectory) Then
             ' Create a DirectoryInfo object representing the specified directory.
-            Dim dir As New DirectoryInfo(directory)
+            Dim dir As New DirectoryInfo(targetDirectory)
             ' Get the FileInfo objects for every file in the directory.
             Dim files As FileInfo() = dir.GetFiles()
             ' Initialize a SHA256 hash object.
             Dim mySHA256 As SHA256 = SHA256Managed.Create()
```

Comment:
SHA256Managed => SHA256; and the VB equivalent of a `using` statement (perhaps just manually calling Dispose later?)